src/GraphQl/Type/TypeBuilder.php

```diff
@@ -84,80 +85,84 @@ public function getResourceObjectType(?string $resourceClass, ResourceMetadataCo
             $shortName .= 'Data';
         }
 
-        if ($this->typesContainer->has($shortName)) {
```

Comment:
quite nice, not sure if reversing this condition was useful (it makes my life harder as the diff is complicated :p) but I think I got it in the end we want to apply the `required` to every resourceObjectType. Maybe we can make this more readable (see next comment)

Reply:
Yes, i did this because otherwise the type would set `set` in the typesContainer as `nonNull` in case of a required type. If later in the same entity there would be a different field that refers to the same entity, it would always get this `nonNull` type even if it's NOT required. Bit of an edge-case, but this way we only 'store' the type, and optionally wrap it into a `nonNull` type based on the ApiProperty settings.